core/master/src/main/java/org/drftpd/commands/usermanagement/expireduser/ExpiredUserManager.java

```diff
@@ -76,14 +78,25 @@ private void doExpired(User user) {
 		if (!_chgrp.isEmpty()) {
 			String[] groups = _chgrp.split(" ");
 			for (String group : groups) {
-				user.toggleGroup(group);	
+				try {
+					user.toggleGroup(user.getUserManager().getGroupByName(group));
+				} catch (NoSuchGroupException e) {
+					logger.warn("[doExpired] Tried to chgrp to a unknown group: {}", group);
+				} catch (GroupFileException e) {
+					logger.warn("There was an error reading the group file for {} while trying to use chgrp", group, e);
+				}
 			}
-			
 		}
 		
 		if (!_setgrp.isEmpty()) {
 			String[] groups = _setgrp.split(" ");
-			user.setGroup(groups[0]);
```

Comment:
We can specify a lists but we dont care?

Reply:
Didn't notice that part when I refactored it. There can only be one primary group per user. I added a warning level log entry if there is more than 1 group configured here